src/lib.rs

```diff
@@ -65,6 +66,12 @@ impl Default for Client {
             .pool_idle_timeout(POOL_IDLE_TIMEOUT)
             .build(connector);
 
+        Self::with_http_client(Box::new(client))
+    }
+}
+
+impl Client {
+    pub fn with_http_client(client: BoxHttpClient) -> Self {
```

Comment:
I think, it's better to accept `impl HttpClient` here and wrap into `Arc<dyn HttpClient>` inside the function to avoid exposing `BoxHttpClient` type.

Reply:
Fixed